backend/scraper/__init__.py

```diff
@@ -0,0 +1,93 @@
+"""Scrape research papers from the Semantic Scholar API and put them into the database."""
+import semanticscholar as ss
+import requests
+
+def scrape():
+    """Scrape research papers from the Semantic Scholar API"""
+    # Determine the ids of all relevant research papers.
+    papers = get_all_papers()
+    for paperId in papers:
+        # Get all relevant information for the paper: id, title, abstract, year
+        paper = ss.paper(paperId)
+        paperTitle = paper['title']
+        paperAbstract = paper['abstract']
+        paperYear = paper['year']
+        # Put the given paper into the database.
+        post_paper(paperId, paperTitle, paperAbstract, paperYear)
+
+        # Get all relevant information for the author: id, name
+        authors = paper['authors']
+        for author in authors:
+            authorId = author['authorId']
+            authorName = author['name']
+            # Put the given author and writing relation into the database.
+            post_author(authorId, authorName)
+            post_write(paperId, authorId)
+
+        # Get all references
+        citations = paper['citations']
+        if citations == []:
+            citations = paper['references']
```

Comment:
this looks kind of weird. First getting `citations` and then `references` if there are no citations. Could you elaborate on why this needs to be done?

Reply:
The SemanticScholar API sometimes calls citations 'citations' and sometimes 'references'.